### PR TITLE
vips 8.12.0

### PR DESCRIPTION
**Pull request by jcupitt**
- update vips version
- don't set --with-magick, it's the default
- add new cgif dependency
- remove unnecessary giflib dependency
- add jpeg-xl dependency

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

File: Formula/vips.rb [hunk 1]
```diff
@@ -11,20 +11,21 @@ class Vips < Formula
...
 
   bottle do
-    sha256 arm64_big_sur: "96192c9244f233b59b24e56bdb5a52ffaae7ad79f35792b512c5956ceadb159a"
-    sha256 big_sur:       "1ff65470a885cb982a651a9aeb76be2188803f58071547a5c677718cb24ab889"
-    sha256 catalina:      "45e3e0ae7564b37ba1398b48465811889b42783e02b6c3491c0d3dabdda9e3ef"
-    sha256 mojave:        "b4aecdd43904e1acee4da8af57da53ebaa5e29f857c10c528226888457aa9a49"
+    sha256 arm64_big_sur: "7465e3684f04ad85463694971862aa3694ec67655b01e8d5d6eef6af22852af9"
+    sha256 big_sur:       "423bdd4aa945fefc6dd01646892c429494620cc942a96f25e2eab72cc8088df0"
+    sha256 catalina:      "8d60363727bb9f48e8733b8d570732734ae727809489547e8ec3c956a72569f2"
+    sha256 mojave:        "49429e971e5550fd703dcdcd3647a0d14c08267fa7e5c2ac4f6e0a402e92fb2f"
```
Comment (carlocab) [inline Formula/vips.rb:17]: Please don't touch the bottle block. (Where did these new checksums even come from?)

File: Formula/vips.rb [hunk 2]
```diff
@@ -56,7 +57,6 @@ def install
     args = %W[
       --disable-dependency-tracking
       --prefix=#{prefix}
-      --with-magick
```
Comment (carlocab) [inline Formula/vips.rb:59]: If I pass `--with-magick`, and `configure` can't find it, does it carry on as if I didn't ask to use magick, or does it error?

**Comment by carlocab** [comment]
By the way, for future reference, please use Homebrew/core style when formatting your commits and commit messages. Here's a quick summary:

> Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.
> 
> For new formulae:
> 
> > At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.
> 
> For existing formulae:
> 
> > The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.
> 
> Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, use the following guidelines to make sure that @BrewTestBot can merge your commits:
> 
> > - One formula per commit; one commit per formula.
> > - Keep merge commits out of the pull request.

File: Formula/vips.rb (hunk 2 above)
Comment (jcupitt) [reply Formula/vips.rb:59]: It's equivalent to `test`, so there's no error. Though arguably there should be.

File: Formula/vips.rb (hunk 1 above)
Comment (jcupitt) [reply Formula/vips.rb:17]: Ooop, sorry. I didn't touch them, it must have been some strange accident. I'll remove this.

Comment (carlocab) [reply Formula/vips.rb]: Ah, these are the bottle checksums from 34f59d481e1b7a175b3ddb2a486c03865481d5fe. You must've started from an outdated branch when you created your PR. Do 
```
git fetch origin && git checkout -b <branch name> origin/master
```
to make sure your PR branch is up-to-date before working on changes.

Or, you can use `brew bump-formula-pr` (and amend the commits it creates automatically) and it should just do the right thing.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**